### PR TITLE
SDK-1982 SKAN Simplification DB Updates

### PR DESCRIPTION
**Pull request by NidhiDixit09**
## Reference
SDK-1982 -- [iOS] SKAN Simplification DB updates .
https://branch.atlassian.net/browse/SDK-1982 

## Summary
* Added parameter for current SKAN Window (skan_postback_index) in /v1/open and /v2/event requests
* Updated function 'shouldCallPostbackForDataResponse' to handle negative conversion values for second and third SKAN Windows.

## Motivation
Support SKAN 4.0 DB Simplication.

## Type Of Change

- [ ] New feature (non-breaking change which adds functionality)

## Testing Instructions
With Every INSTALL/OPEN/EVENT requests, Verify using logs SDK is sending current SKAN window number. 
Verify in response , SDK is getting back conversion values.
Some of the samples are here -
https://www.notion.so/brancheng/SDK-SKAN-Testing-Results-5a74041cd815457292725673097f66c4

cc @BranchMetrics/saas-sdk-devs for visibility.

### CODE REVIEW

File: BranchSDK/BranchEvent.m [hunk 1]
```diff
@@ -351,6 +351,19 @@ - (NSDictionary *)buildEventDictionary {
         eventDictionary[BRANCH_REQUEST_KEY_PARTNER_PARAMETERS] = partnerParameters;
     }
     
+    if (@available(iOS 16.1, *)){
```
Comment (echo-branch) [inline BranchSDK/BranchEvent.m:354]: Can we add a maccatalyst check as well.

File: BranchSDK/BNCSKAdNetwork.m [hunk 2]
```diff
@@ -183,7 +183,11 @@ - (BOOL) shouldCallPostbackForDataResponse:(NSDictionary *) dataResponseDictiona
     }
     
     int highestConversionValue = (int)[BNCPreferenceHelper sharedInstance].highestConversionValueSent;
-    if( conversionValue.intValue <= highestConversionValue ){
+    if((currentWindow == BranchSkanWindowFirst) && (conversionValue.intValue <= highestConversionValue)){
+        BOOL ascendingOnly = [self getAscendingOnlyFromDataResponse:dataResponseDictionary];
+        if (!ascendingOnly)
+            shouldCallUpdatePostback = YES;
+    } else if ((currentWindow != BranchSkanWindowFirst) && (highestConversionValue != 0) && (conversionValue.intValue <= highestConversionValue)){
```
Comment (echo-branch) [inline BranchSDK/BNCSKAdNetwork.m:190]: Is this second check just to prevent calling more than once with conversion value 0?

Comment (NidhiDixit09) [reply BranchSDK/BNCSKAdNetwork.m:190]: @echo-branch  Now second and Third windows can have negative values (-1 (high), -2(medium), -3(low)) and default value for highestConversionValue (sent so for - used for ascendingOnly feature) is 0 for each window which is always higher than those negative conversion values. So in that scenario, if window is second or third and highest conversion value is 0 , it should go to last `else` condition and save that value in preferences.

File: BranchSDK/BranchEvent.m (hunk 1 above)
Comment (NidhiDixit09) [reply BranchSDK/BranchEvent.m]: @echo-branch Done.

File: BranchSDK/BNCSKAdNetwork.m (hunk 2 above)
Comment (echo-branch) [reply BranchSDK/BNCSKAdNetwork.m]: So the second check is cause we default `highestConversionValueSent` to 0, which would prevent negative conversion values from being saved. What if we default it to `NSIntegerMin` instead?

**Review by echo-branch — APPROVED** [review]

**Review by echo-branch — APPROVED** [review]

**Pull request merged**